File: March_2024/D5/Solution.py
class Solution(object):
    def minimumLength(self, s):
        """
        :type s: str
        :retype: int
        """
        left = 0
        right = len(s) - 1
        while left < right and s[left] == s[right]:
            temp = s[left]
            while left < len(s) and s[left] == temp:
                left += 1
            while right > -1 and s[right] == temp:
                right -= 1
            if right < left:
                return 0
        return right - left + 1

File: March_2024/D5/Solution.java
package March_2024.D5;

public class Solution {
    public static void main(String[] args) {

    }

    public int minimumLength(String s) {
        int left = 0, right = s.length() - 1;
        while (right > left && s.charAt(left) == s.charAt(right)) {
            int temp = left;
            while (left < s.length() && s.charAt(left) == s.charAt(temp)) {
                left++;
            }
            while (right > -1 && s.charAt(right) == s.charAt(temp)) {
                right--;
            }
            if (right < left) {
                return 0;
            }
        }
        return right - left + 1;
    }
}
